### ISSUE TITLE
The font size of a label with formattedtext is not the same as the fontsize of a label with regular text

### ISSUE THREAD

**Issue by sjorsmiltenburg** [dotnet/maui#18098]
### Description

Occurs on Windows, not yet tested other platforms
![image](https://github.com/dotnet/maui/assets/649207/13e4a369-e157-4e4d-af1a-3cb606233a1f)

### Steps to Reproduce

open sample project - press bug9 button

### Link to public reproduction project repository

https://github.com/sjorsmiltenburg/maui_bugs.git

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows

### Did you find any workaround?

no

**Comment by SmartmanApps** [comment]
Yes! I have just recently run into this, having just used formatted text for the first time. Thought that was odd.

I also want to add (haven't created an issue yet because I haven't created a repro repo yet) that on Windows (not on Android - works on Android) trailing space is being ignored in the label's text. So, quick little fix to make the background a bit bigger, add a space either side of the text, but on Windows the trailing space gets ignored. That's even with regular text, not just formatted text.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.8.0 Preview 4.0(8.0.0-rc.2.9373). Repro on Windows 11, Android 13.0-API33 and iOS 16.4 with below Project:
[maui_bugs.zip](https://github.com/dotnet/maui/files/13038412/maui_bugs.zip)

**Comment by Dhivya-SF4094** [comment]
The issue occurs because Span elements inside a FormattedString do not inherit the FontFamily from their parent Label when styles are applied via styles.xaml.
Since styles.xaml defines a FontFamily only for Label and not for Span, the Span elements fallback to the system default font, leading to inconsistencies in rendering.
As a workaround, explicitly set the FontFamily for each Span inside the FormattedString.

[MauiApp1 3.zip](https://github.com/user-attachments/files/19630196/MauiApp1.3.zip)
